tests/batch/test_FileNameRetriever.py

```diff
@@ -0,0 +1,195 @@
+import pytest
+from dfastmi.batch import FileNameRetriever
+from configparser import ConfigParser
+
+class Test_FileNameRetriever():
+    class Test_FileNameRetrieverFactory():
+        @pytest.fixture
+        def factory(self) -> FileNameRetriever.FileNameRetrieverFactory:
+            factory = FileNameRetriever.FileNameRetrieverFactory()
+            factory.register_creator("1.0", lambda needs_tide: FileNameRetriever.FileNameRetrieverLegacy())
+            factory.register_creator("2.0", lambda needs_tide: FileNameRetriever.FileNameRetriever(needs_tide))
+            return factory
+        
+        def given_version_1_with_varying_needs_tide_when_generate_then_return_FileNameRetrieverLegacy(self, factory : FileNameRetriever.FileNameRetrieverFactory):
```

Comment:
Unclear because I forgot.. but wasn't it agreed between us to add a text here splitting the name in GWT:
```suggestion
     """"
     given : version 1 with varying needs tide
     when :  generate 
     then  : FileNameRetrieverLegacy
     """
```